### PR TITLE
Structured reservoirs

### PR DESCRIPTION
**Pull request by hellkite500**
Integrate reservoirs into `compute_network_structured`

## Additions

- Backend C libs for reach strcutures

## Changes

- Add some reach identity management for reservoir reaches

## Todos

- Might need to clean up the cython wrappers a bit, and maybe standardize some of the reach interfaces

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [x] MacOS
- [x] Linux

### CODE REVIEW

**Comment by jdmattern-noaa** [comment]
I plan to go through this on Wed.

**Review by jdmattern-noaa — APPROVED** [review]
Great work, Nels!

File: src/python_framework_v02/troute/network/reach_structs.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#ifndef REACH_STRUCTS_H
+#define REACH_STRUCTS_H
+/*
+    C Structures
+*/
+#include "musking/mc_reach_structs.h"
+#include "reservoirs/levelpool/levelpool_structs.h"
```
[the remark this answers is not in the document]
Comment (hellkite500) [reply src/python_framework_v02/troute/network/reach_structs.h:7]: Not with the #define include guards on the headers.

**Comment by hellkite500** [comment]
Closing in favor of #263 -- same code, cleaner rebase, more bug fixes

**Pull request closed without merging**